### DISCUSSION TITLE
Image optimization causes an error in displaying the fetched images from strapi

### DISCUSSION THREAD

**Discussion by alirezahekmati** [vercel/next.js#39239] [Help]
### problem⛔:
 image URLs fetched from strapi API don't display correctly on the page. 
### reason for the problem⁉:
 I find out it happens because of image optimization so when you do unoptimized={true} it will work fine. 
```
<Image
            unoptimized={true}
            src={`http://localhost:1337${image URL from api}`}
            width={400}
            height={400}
            alt="an image from starpi API"
          />
```
### the goal✅: 
how to keep image optimization when working with API from strapi.

#### 1.strapi :

1.1. create a strapi project 
 

> npx create-strapi-starter my-project next-blog.

1.2. add a content type that includes images.
1.3. add images to the strapi database.

#### 2.next js:

2.1. create nextjs project 
     

> npx create-next-app@latest

2.2. fetch image data from starpi API 
     

> http://localhost:1337/api/<API ID (Singular)>?populate=*

2.3. use the fetched URL in Image component

```
<Image
            src={`http://localhost:1337${image URL from api}`}
            width={400}
            height={400}
            alt="an image from starpi API"
          />
```

### more info 🟢:
everything works fine **as long as you use  unoptimized={true}.**
btw I have added localhost to the config file:
```
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  swcMinify: true,
  images: {
    domains: ["localhost:1337", "localhost"],
  },
};

module.exports = nextConfig;
```

**Comment by alirezahekmati** [comment]
the problem will be fixed when you host strapi to a VPS.✅
basically, if you run strapi and next in the same server image optimization will throw an error.

**Comment by nischolas** [comment]
@alirezahekmati What do you mean by "images don't display correctly on the page"?

I'm having a similar problem. Next/image adds "?w=3840&q=75" to every image url and Strapi doesn't seem to like the URL parameters (responds with 404). `unoptimized={true}` fixes it but I want to avoid disabling the image optimization.

I'm going to push it to my VPS and check if that helps.
